exporter/prometheusremotewriteexporter/wal_test.go

```diff
@@ -156,3 +159,64 @@ func TestWAL_persist(t *testing.T) {
 	require.Equal(t, reqLFromWAL[0], reqL[0])
 	require.Equal(t, reqLFromWAL[1], reqL[1])
 }
+
+func TestWAL_E2E(t *testing.T) {
+	in := []*prompb.WriteRequest{
+		series("mem_used_percent", 0, 0),
+		series("mem_used_percent", 15, 34),
+		series("mem_used_percent", 30, 99),
+	}
+	out := make([]*prompb.WriteRequest, 0, len(in))
+
+	done := make(chan struct{})
+	sink := func(ctx context.Context, reqs []*prompb.WriteRequest) error {
+		out = append(out, reqs...)
+		if len(out) >= len(in) {
+			close(done)
+		}
+		return nil
+	}
+
+	wal, err := newWAL(&WALConfig{
+		Directory: t.TempDir(),
+	}, sink)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(t.TempDir())
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	log, err := zap.NewDevelopment()
+	if err != nil {
+		panic(err)
+	}
```

Comment:
You shouldn't need another logger here, I suggest you create one at the start of the test and use that throughout.